### PR TITLE
Use ipc router in webdriver

### PR DESCRIPTION
**Pull request by gterzian**
Fix for https://github.com/servo/servo/issues/23905#issuecomment-522588119

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/24007)

### CODE REVIEW

**Comment by gterzian** [comment]
@asajeffrey r?

File: components/webdriver_server/lib.rs [hunk 1]
```diff
@@ -612,35 +622,26 @@ impl Handler {
...
-
-        // wait to get a load event
-        match receiver.recv().unwrap() {
-            LoadStatus::LoadComplete => Ok(WebDriverResponse::Void),
-            LoadStatus::LoadTimeout => {
-                Err(WebDriverError::new(ErrorStatus::Timeout, "Load timed out"))
-            },
+        select! {
+            recv(self.load_status_receiver) -> _ => Ok(WebDriverResponse::Void),
+            recv(after(Duration::from_millis(timeout))) -> _ => Err(
```
Comment (Manishearth) [inline components/webdriver_server/lib.rs]: beautiful

**Review by asajeffrey — APPROVED** [review]
Looks good, the only thing is it could do with some comments, In particular, it would be nice to have some text about why we have one ipc and one crossbeam channel.

**Review by Manishearth — APPROVED** [review]

**Comment by gterzian** [comment]
@asajeffrey Ok I have added comments and rebased...

File: components/webdriver_server/lib.rs [hunk 2]
```diff
@@ -171,6 +174,14 @@ impl WebDriverSession {
 }
 
 struct Handler {
+    /// The threaded receiver on which we can block for a load-status.
+    /// It will receive messages sent on the load_status_sender,
+    /// and forwarded by the IPC router.
+    load_status_receiver: Receiver<LoadStatus>,
+    /// The IPC sender which we can clone and pass along to the constellation,
+    /// for it to send us a load-status. Messages sent on it
+    /// will be forward to the load_status_receiver.
```
Comment (CYBAI) [inline components/webdriver_server/lib.rs]: `s/forward/forwarded` 👀  ?

Comment (gterzian) [reply components/webdriver_server/lib.rs]: thanks

**Comment by asajeffrey** [comment]
Might be worth saying that we're doing the forwarding because IPC channels don't support recv_timeout.

**Comment by gterzian** [comment]
Ok, I've added a comment to that effect.

**Comment by asajeffrey** [comment]
Thanks! @bors-servo r+

**Pull request merged**